### ISSUE TITLE
 OSError: cannot load library 'gobject-2.0-0': gobject-2.0-0: cannot open shared object file: No such file or directory.  Additionally, ctypes.util.find_library() did not manage to locate a library called 'gobject-2.0-0'

### ISSUE THREAD

**Issue by Henoker** [Kozea/WeasyPrint#1772]
Hello, I am trying use weasyprint pdf generator in Dockerized Django project. I've installed weasyprint as follows using my dockerfile. 
`# Pull base image
FROM python:3.10.4-slim-bullseye
ENV WEASYPRINT_VERSION 57.1
RUN pip install weasyprint==$WEASYPRINT_VERSION
# Set environment variables
ENV PIP_DISABLE_PIP_VERSION_CHECK 1
ENV PYTHONDONTWRITEBYTECODE 1
ENV PYTHONUNBUFFERED 1
# Set work directory
WORKDIR /code
# Install dependencies
COPY ./requirements.txt .

RUN pip install -r requirements.txt
# Copy project
COPY . .
EXPOSE 8000
all` 

However. after I successfully installed weasyprint by using docker-compose up -d --build docker-compose.yml command. My localhost didnt load. When I run docker-compose logs command I receive:   OSError: cannot load library 'gobject-2.0-0': gobject-2.0-0: cannot open shared object file: No such file or directory.  Additionally, ctypes.util.find_library() did not manage to locate a library called 'gobject-2.0-0' error. Can anyone help me on figuring out this issue?

**Comment by liZe** [comment]
Hi!

As you use a Debian image, you can follow [the installation guide for Debian](https://doc.courtbouillon.org/weasyprint/stable/first_steps.html#debian-11).

**Comment by Henoker** [comment]
Yes, I am using debian, I followed the instruction by adding in docker file the following command:
`RUN apt-get update -y && apt-get install python3-pip python3-cffi python3-brotli libpango-1.0-0 libpangoft2-1.0-0`
However, I am now facing another error:

 Need to get 122 MB of archives.
After this operation, 430 MB of additional disk space will be used.
Do you want to continue? [Y/n] Abort.
_________
executor failed running [/bin/sh -c apt-get update -y && apt-get install python3-pip python3-cffi python3-brotli libpango-1.0-0 libpangoft2-1.0-0]: exit code: 1

**Comment by liZe** [comment]
As you’re not using an interactive CLI, you have to avoid the confirmation question. You can use `apt-get install`’s `-y` option to force the confirmation.

**Comment by Henoker** [comment]
Thanks you very much. The debian link solves the first issue.
For the second issue, I have added the following command:
FROM python:3.10.4-slim-bullseye
RUN apt-get update
RUN apt-get -y install python3-pip python3-cffi python3-brotli libpango-1.0-0 libpangoft2-1.0-0
ENV WEASYPRINT_VERSION 57.1
RUN pip install weasyprint==$WEASYPRINT_VERSION
# Set environment variables
ENV PIP_DISABLE_PIP_VERSION_CHECK 1
ENV PYTHONDONTWRITEBYTECODE 1
ENV PYTHONUNBUFFERED 1
# Set work directory
WORKDIR /code

# Install dependencies
COPY ./requirements.txt .

RUN pip install -r requirements.txt
# Copy project
COPY . .
EXPOSE 8000

Now it works fine in docker without virtual environment.

**Comment by dedalusMohantyMa** [comment]
btw, for alpine image: `RUN apk add pango libffi` is enough

**Comment by ejmmanning76** [comment]
dose anyone know what kind of image fly.io uses I am having the same problem and I do not know docker very well
